pkg/webhooks/admission/jobs/validate/admit_job.go

```diff
@@ -226,9 +226,30 @@ func validateJobCreate(job *v1alpha1.Job, reviewResponse *admissionv1.AdmissionR
 	queue, err := config.VolcanoClient.SchedulingV1beta1().Queues().Get(context.TODO(), job.Spec.Queue, metav1.GetOptions{})
 	if err != nil {
 		msg += fmt.Sprintf(" unable to find job queue: %v;", err)
-	} else if queue.Status.State != schedulingv1beta1.QueueStateOpen {
-		msg += fmt.Sprintf(" can only submit job to queue with state `Open`, "+
-			"queue `%s` status is `%s`;", queue.Name, queue.Status.State)
+	} else {
+		if queue.Status.State != schedulingv1beta1.QueueStateOpen {
+			msg += fmt.Sprintf(" can only submit job to queue with state `Open`, "+
+				"queue `%s` status is `%s`;", queue.Name, queue.Status.State)
+		}
+
+		// valiadate hierarchical queue
+		if queue.Name == "root" {
+			msg += " can not submit job to root queue;"
+		} else {
+			queueList, err := config.VolcanoClient.SchedulingV1beta1().Queues().List(context.TODO(), metav1.ListOptions{})
+			if err != nil {
+				msg += fmt.Sprintf("failed to get list queues: %v;", err)
+			}
+			childQueues := make([]schedulingv1beta1.Queue, 0)
+			for _, childQueue := range queueList.Items {
+				if childQueue.Spec.Parent == queue.Name {
+					childQueues = append(childQueues, childQueue)
+				}
+			}
+			if len(childQueues) > 0 {
+				msg += fmt.Sprintf(" can only submit job to leaf queue, "+"queue `%s` has %d child queues;", queue.Name, len(childQueues))
+			}
```

Comment:
I don't think it's necessary to go through all the queues here. As long as there is a childQueue, an error will be returned

Reply:
I have already addressed the other issues you mentioned above. Regarding this particular issue, I would prefer to return all the error information at once, so that the user can resolve all the problems in one go based on the error messages. It's also possible to modify the code as you suggest, but that might lead to the following scenario: when a user is using the system, there are actually three erroneous queues, but only one error queue is displayed each time a change is made, requiring repeated modifications.